services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/extensions/dynamodb/mappingclient/core/DynamoDbAsyncMappedDatabaseTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.extensions.dynamodb.mappingclient.core;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.DatabaseOperation;
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.MapperExtension;
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.TableSchema;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DynamoDbAsyncMappedDatabaseTest {
+    @Mock
+    private DynamoDbAsyncClient mockDynamoDbAsyncClient;
+    @Mock
+    private MapperExtension mockMapperExtension;
+    @Mock
+    private DatabaseOperation<?, ?, String> mockDatabaseOperation;
+    @Mock
+    private TableSchema<Object> mockTableSchema;
+
+    @InjectMocks
```

Comment:
I use it when it's convenient, when it breaks (because you add a new dependency without mocking it in your test) in TDD you normally write a test first for that new dependency and so the silent failure hazard doesn't really apply. I'm going to leave as-is, but it's a good call-out and something to be aware of when using it.